packages/contracts/contracts/L1/Colosseum.sol

```diff
@@ -687,15 +708,16 @@ contract Colosseum is Initializable, ISemver {
         if (_outputRoot == DELETED_OUTPUT_ROOT) revert CannotRollbackOutputToZero();
         if (L2_ORACLE.getL2Output(_outputIndex).outputRoot != DELETED_OUTPUT_ROOT)
             revert OutputNotDeleted();
-        verifiedPublicInputs[_publicInputHash] = false;
 
         // Rollback output root.
-        L2_ORACLE.replaceL2Output(_outputIndex, _outputRoot, _asserter);
+        L2_ORACLE.replaceL2Output(_outputIndex, _outputRoot, SECURITY_COUNCIL);
```

Comment:
As I understand, this solution to replace the submitter to security council in the case of ZK bug gives
- challenge reward to asserter
- output reward to security council
- no bond back to the asserter
Is this right? Then I think it's somewhat unfair to give the output reward to security council, since it was actually the fund that should go to the asserter if there was no ZK related bug. How about making an extra function like `recoverAsserterBond()` that recovers asserter's bond back inside the below if statement?

Reply:
Yes, what you listed is almost correct, but challenge reward will also be transferred to security council.

In this way, we need to refund the asserter's slashed bond + challenge reward + output reward to the asserter manually. In V1, we also need to refund the tax to the asserter manually in dismissing challenge case.

I also considered adding `recoverAsserterBond`, but the tax amount is already tranferred to security council during slashing, so we cannot recover the asserter's bond fully. But in output finalization, the full bond amount should be unbonded, so there are also some difficulties in that way.

Reply:
> Yes, what you listed is almost correct, but challenge reward will also be transferred to security council.

Oh yeah that's right.

How about transferring tax back to the `AssetManager` from security council in the case of `dismissChallenge`? Is there any context at the time we decide not to return the tax back to `ValidatorPool` at V1?

Reply:
There was no context at that time, since we did not actually undergo such cases.

Hmm then how about transferring the tax at the time of output finalization? Just add `BOND_AMOUNT` to `_pendingChallengeReward` and handle it after all the challenges are concluded.

Reply:
@0xHansLee wdyt about this issue?

Reply:
That seems better imo, how do you think @0xHansLee ?

Reply:
I applied it, please take a look.

Reply:
It's reasonable to me. 
BTW, when the challenge is dismissed, should the asserter be refunded? Once the challenge is proven, the asserter get slashed. I think asserter should be refunded the bond that already slashed.

Reply:
Oh I missed the part. I'll add `recoverAsserterBond` function.